soot-infoflow/pom.xml

```diff
@@ -126,34 +145,127 @@
 			<scope>test</scope>
 		</dependency>
 		<dependency>
-			<groupId>java.j2ee</groupId>
-			<artifactId>j2ee</artifactId>
-			<version>1.4</version>
-			<scope>test</scope>
+		    <groupId>javax.servlet</groupId>
+		    <artifactId>javax.servlet-api</artifactId>
+		    <version>3.0.1</version>
+		    <scope>test</scope>
 		</dependency>
 		<dependency>
 		    <groupId>net.sf.trove4j</groupId>
 		    <artifactId>trove4j</artifactId>
 		    <version>3.0.3</version>
 		</dependency>
-		<!-- https://mvnrepository.com/artifact/com.google.guava/guava -->
 		<dependency>
 		    <groupId>com.google.guava</groupId>
 		    <artifactId>guava</artifactId>
 		    <version>25.1-jre</version>
 		</dependency>
 	</dependencies>
 
-	<repositories>
-		<repository>
-			<id>soot-snapshot</id>
-			<name>Soot snapshot server</name>
-			<url>https://soot-build.cs.uni-paderborn.de/nexus/repository/soot-snapshot/</url>
-		</repository>
-		<repository>
-			<id>soot-release</id>
-			<name>Soot release server</name>
-			<url>https://soot-build.cs.uni-paderborn.de/nexus/repository/soot-release/</url>
-		</repository>
-	</repositories>
+<repositories>
+        <repository>
+            <id>sonatype-snapshot</id>
+            <url>https://oss.sonatype.org/content/repositories/snapshots</url>
+            <releases>
+                <enabled>false</enabled>
+            </releases>
+            <snapshots>
+                <enabled>true</enabled>
+            </snapshots>
+        </repository>
+    </repositories>
+    <distributionManagement>
+        <snapshotRepository>
+            <id>ossrh</id>
+            <url>https://oss.sonatype.org/content/repositories/snapshots</url>
+        </snapshotRepository>
+    </distributionManagement>
+    <profiles>
+        <profile>
+            <id>deploy</id>
+            <activation>
+                <property>
+                    <name>deploy</name>
+                    <value>true</value>
+                </property>
+            </activation>
+            <properties>
+                <gpg.executable>gpg2</gpg.executable>
+                <gpg.keyname>BA6398DCF4E2C1B6C5C7BDFC364F80C21BCC61BB</gpg.keyname>
```

Comment:
Please keep references to signing keys out of the POM file and rather specify them in your local build configuration. Otherwise, you break the build for everyone else.

Reply:
Any proposal for a clean solution?

Reply:
I think you might be able to inject the property into maven in the build job itself.
However, the whole deploy profile would fail even without a gpg key due to failing to push to Maven Central. Tbh, I would just keep it as is and maybe put source/jdoc generation in another profile if required. Everyone else should not try to deploy FlowDroid to Maven Central anyway.

Reply:
I guess we can move it to the build system's `mvnsettings.xml`

Reply:
Moving the pgp key info into the build system's `mvnsettings.xml` works. :+1:

Reply:
I would appreciate this solution. At Fraunhofer SIT, we must also build FlowDroid and deploy it to our own artifact servers, e.g., for testing builds that are not stable enough yet for the public repo. Therefore, I'd appreciate server-specific configurations for PGP keys and the like.

Reply:
Okay, I see. I've moved the pgp key info out of the `pom.xml` in commit 58f167b.